Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift

```diff
@@ -66,38 +42,74 @@ import BraintreeCore
+    ///    - amount: The amount for the transaction.
+    ///    - billingAddress: Optional. The billing address used for verification
+    ///    - cardAddChallengeRequested: Optional.  An authentication created using this flag should only be used for vaulting operations (creation of customers' credit cards or payment methods) and not for creating transactions. If set to `true`, a card-add challenge will be requested from the issuer. If set to `false`, a card-add challenge will not be requested. If the parameter is missing, a card-add challenge will only be requested for $0 amount.
+    ///    - challengeRequested: Optional. If set to true, an authentication challenge will be forced if possible.
+    ///    - customFields: Object where each key is the name of a custom field which has been configured in the Control Panel. In the Control Panel you can configure 3D Secure Rules which trigger on certain values.
+    ///    - dataOnlyRequested:  Optional. Indicates whether to use the data only flow. In this flow, frictionless 3DS is ensured for Mastercard cardholders as the card scheme provides a risk score for the issuer to determine whether to approve. If data only is not supported by the processor, a validation error will be raised. Non-Mastercard cardholders will fallback to a normal 3DS flow.
+    ///    - dfReferenceID: The dfReferenceID for the session. Exposed for testing.
+    ///    - email: Optional. The email used for verification.
+    ///    - exemptionRequested: Optional. If set to true, an exemption to the authentication challenge will be requested.
+    ///    - mobilePhoneNumber: Optional. The mobile phone number used for verification. Only numbers. Remove dashes, parentheses and other characters.
+    ///    - nonce: A nonce to be verified by ThreeDSecure
+    ///    - renderTypes: Optional: List of all the render types that the device supports for displaying specific challenge user interfaces within the 3D Secure challenge. When using `BTThreeDSecureUIType.both` or `BTThreeDSecureUIType.html`, all `BTThreeDSecureRenderType` options must be set. When using `BTThreeDSecureUIType.native`, all `BTThreeDSecureRenderType` options except `.html` must be set.
+    ///    - requestedExemptionType: Optional. The exemption type to be requested. If an exemption is requested and the exemption's conditions are satisfied, then it will be applied.
+    ///    - shippingMethod: Optional. The shipping method chosen for the transaction
+    ///    - uiType: Optional: Sets all UI types that the device supports for displaying specific challenge user interfaces in the 3D Secure challenge. Defaults to `.both`
+    ///    - v2UICustomization: Optional. UI Customization for 3DS2 challenge views.
+    public init(
+        accountType: BTThreeDSecureAccountType = .unspecified,
+        additionalInformation: BTThreeDSecureAdditionalInformation? = nil,
+        amount: NSDecimalNumber? = 0,
```

Comment:
Based on this logic it looks like both amount and the delegate may be required: https://github.com/braintree/braintree_ios/blob/5ecb18dc594019e5bf9292fd67871b8f15812fd1/Sources/BraintreeThreeDSecure/BTThreeDSecureClient.swift#L62-L74

Reply:
ahh yep, good catch! the amount does seem to be required (i've also confirmed this by raising a BT 3D Secure General Inquiry workflow request) - as for the delegate, it seems that it was also previously optional so i wonder if it's look to leave that as is 👀